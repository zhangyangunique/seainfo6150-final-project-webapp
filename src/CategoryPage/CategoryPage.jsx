import React from "react";
import styles from "./CategoryPage.module.css";
import ArticleList from "../Article/ArticleList/ArticleList.jsx";
import articlesData from "../data/articles.json"
import { isEmpty } from "lodash";

const CategoryPage = (props) => {
	const fetchedData = articlesData[props.categoryId] === undefined ? null : Object.values(articlesData[props.categoryId]);

	return isEmpty(fetchedData) ? null : (
		<div className={styles.container}>
			<div className={styles.mainbody}>
				<h2>{props.categoryId} Articles</h2>
				<ArticleList articles={fetchedData} />
			</div>
		</div>
	)
}

export default CategoryPage;
