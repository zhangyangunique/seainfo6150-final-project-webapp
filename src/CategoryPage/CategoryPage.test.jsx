import React from "react";
import { BrowserRouter } from 'react-router-dom';
import { render } from "@testing-library/react";
import CategoryPage from "./CategoryPage";

const categoryId = "sport";

describe("CategoryPage tests", () => {
    it("renders correctly", () => {
        const { container } = render(
            <BrowserRouter>
                <CategoryPage categoryId={categoryId} />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});