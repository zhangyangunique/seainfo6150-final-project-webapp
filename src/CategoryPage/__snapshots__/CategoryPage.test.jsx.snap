// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="mainbody"
    >
      <h2>
        sport
         Articles
      </h2>
      <div>
        <ul
          class="container"
        >
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="In Winning His Fourth NBA Title, LeBron James Makes an All-Time Statement"
                class="image"
                src="https://api.time.com/wp-content/uploads/2020/10/GettyImages-1279730544.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/181982122240-winningnba"
                  >
                    In Winning His Fourth NBA Title, LeBron James Makes an All-Time Statement
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    The legend wins his fourth title with his third different franchise.
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="'Undoubtedly' the Greatest: Why the World Is Mourning Soccer Legend Diego Maradona"
                class="image"
                src="https://api.time.com/wp-content/uploads/2020/11/AP20331376443033.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/181112124440-sporpaper"
                  >
                    'Undoubtedly' the Greatest: Why the World Is Mourning Soccer Legend Diego Maradona
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    The Argentine soccer legend fought demons off the field, but was an artist on the pitch.
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="Cost to Delay Tokyo Olympics Is Estimated at $2 Billion"
                class="image"
                src="https://api.time.com/wp-content/uploads/2020/11/Olympic.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/181386122240-delaytokyoolympics"
                  >
                    Cost to Delay Tokyo Olympics Is Estimated at $2 Billion
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    The official cost of putting on the Tokyo Olympics is $12.6 billion
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="College Football Feels All Too Normal During the Pandemic"
                class="image"
                src="https://api.time.com/wp-content/uploads/2020/10/arkansas-mississippi-college-football-fans-coronavirus.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/181119572240-collegefootball"
                  >
                    College Football Feels All Too Normal During the Pandemic
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    Universities are subjecting unpaid athletic workers to precisely the same health risks that are eliciting such outrage every day on network television and social media.
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="Cliff-jumping off Ireland's Atlantic coast"
                class="image"
                src="https://cdn.cnn.com/cnnnext/dam/assets/181112122240-coasteering-3-super-169.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/181112122240-coasteering"
                  >
                    Cliff-jumping off Ireland's Atlantic coast
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    If someone told you to jump into an often tumultuous and always frigid North Atlantic Ocean from a 30-foot-high cliff in Ireland, would you do it?
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="31 beautiful reasons to visit England"
                class="image"
                src="https://cdn.cnn.com/cnnnext/dam/assets/160418164324-beautiful-england-29-bath2-super-169.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/160418164324-beautiful-england"
                  >
                    31 beautiful reasons to visit England
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    Discover England's most beautiful places. From the wild North York Moors to the ancient Roman sites in Bath, you be enchanted by scenes natural and man-made.
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="The 32 most beautiful places to visit in Ireland"
                class="image"
                src="https://cdn.cnn.com/cnnnext/dam/assets/160315093137-tollymore-forest-super-169.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/160315093137-beautiful-ireland"
                  >
                    The 32 most beautiful places to visit in Ireland
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    From the dramatic Cliffs of Moher to regal Powerscourt Estate in Wicklow, escape to 32 of the most beautiful places to visit in Ireland.
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="Boeing's 777X jetliner comes together "
                class="image"
                src="https://cdn.cnn.com/cnnnext/dam/assets/181121164246-boeing-777-9x-assembly-1-super-169.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/181121164246-boeing-777-9x-assembly-1"
                  >
                    Boeing's 777X jetliner comes together 
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    Travelers around the globe are one step closer to seeing a plane with its wings folded up being taxied to their gate.
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="container"
          >
            <article
              class="article"
            >
              <img
                alt="Training India's next generation of female pilots"
                class="image"
                src="https://cdn.cnn.com/cnnnext/dam/assets/181118211616-banasthali-vidyapith-aviation-school-india---group-super-169.jpg"
              />
              <div
                class="wrapper"
              >
                <h2
                  class="title"
                >
                  <a
                    href="/articles/181118211616-banasthali-vidyapith-aviation-school-india"
                  >
                    Training India's next generation of female pilots
                  </a>
                </h2>
                <div
                  class="text"
                >
                  <p>
                    On a dusty, 3,600-foot-long airstrip in Rajasthan, India, a two-seater white Cessna 152 pivots into position. A few moments later, it's soaring over the empty desert into a cloudless blue sky.
                  </p>
                </div>
              </div>
            </article>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
